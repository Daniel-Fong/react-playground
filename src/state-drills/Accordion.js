import React from 'react';

class Accordion extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
        index: 0
    }
}
    static defaultProps = { sections: []};

    handleClick () {
        debugger
        this.setState({state: 3})
    }

    render = () => {
        const lis = this.props.sections.map((section, index) => 
                <li key={index}>
                    {this.props.sections.length && (
                        <div>
                            {() => this.state.index}
                            <button onClick={() => this.handleClick()}>{section.title}</button>
                            <p className='isClosed'></p>
                        </div>
                    )} 
                </li>
        )
        return (
            <ul>
                {lis}
            </ul>
        )
    }   
}

export default Accordion;