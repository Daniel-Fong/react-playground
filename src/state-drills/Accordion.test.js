import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import renderer from 'react-test-renderer';

describe(`Accordion`, () => {
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  it('renders empty li when sections not given', () => {
    const tree = renderer.create(<Accordion section={[]} />);
    expect(tree).toMatchSnapshot()
  })
})