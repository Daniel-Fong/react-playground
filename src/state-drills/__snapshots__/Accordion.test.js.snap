// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion renders empty li when sections not given 1`] = `<ul />`;
